### DISCUSSION TITLE
AOSP 12.1 v412

### DISCUSSION THREAD

**Discussion by phhusson** [phhusson/treble_experimentations#2260] [General]
- Make "disable A2DP offload" more robust and work on more devices (fixes A2DP on Galaxy S22)
- Upgrade to April Security Patch Level
- Fix fingerprint (detection &  cleanup) for some devices
- Disable Samsung recovery override
- Possibly fix audio for some cases
- Fix vibrator on Xiaomi Android 12 vendors

<hr /><em>This discussion was created from the release <a href='https://github.com/phhusson/treble_experimentations/releases/tag/v412'>AOSP 12.1 v412</a>.</em>

**Comment by pranav-vaid** [comment]
@phhusson hello,
Thanks for the new build. I just wanted to ask if this release will be available to AOSP 12.1 v411 via OTA update?

**Reply by phhusson** [reply]
Yes, but I usually wait a few days before putting to OTA, so that I'm confident there is no regression.

**Reply by pranav-vaid** [reply]
Yes, I understand that there will be days before any update comes through OTA channels for any rom.
Thanks for the clarification.

**Reply by pranav-vaid** [reply]
@phhusson any update on this?

**Reply by pranav-vaid** [reply]
@phhusson I see that you've added it on OTA channel. Thank you. But I have a doubt. My security patch is still showing 1 Jan 2022 in updater settings, is it okay or that is an error?

**Reply by phhusson** [reply]
If you're using secure variant, it's normal. Secure variants lies, to make it look like you're on stock rom. So it fakes the security patch it shows to match your stock rom.

You can still check that buildid in settings is april 

If you're not, maybe the OTA failed, or maybe you're using magisk which breaks ota.

**Reply by pranav-vaid** [reply]
Gotcha. I am indeed on secure variant. And build is from april. So everything seems good. Thank you.

**Comment by Dunedan** [comment]
Is Monet theming included in this build? If yes: How to configure it?

**Comment by Dunedan** [comment]
I'm happy to announce that this release fixes the long-standing bug of broken A2DP for certain Motorola devices (https://github.com/phhusson/treble_experimentations/issues/1404). :tada:

**Comment by fob014** [comment]
#1 no support for volte
#2 4g voice calling not working (( even after installation  IMS app))
#3 Fingerprint when I open set Fingerprint in security settings than whole setting app stop working and I automatically back to home  
Device = Moto g9 Android 11 stock earlier
Gsi  = i install system-squeak-arm64-ab-vndklite-gapps-secure.img.xz

Meanwhile rom is  very good I like it much please fix this issue

**Reply by luiz-gadelha** [reply]
this happens on mine too, my device is lg k41s,this bug started in v411 version already in v402 it works normally

**Reply by albertordzcz** [reply]
does volte issue was fixed on the  v412 build?

**Comment by Keviny9** [comment]
I have lockscreen bug, I think that they is related for Vendor, but I haven't this bug in A10 and A11
Full Log:
[log1.txt](https://drive.google.com/file/d/1EbsmEXfvLpVThY8k3q0-hG7quEUlsQTq/view?usp=sharing)

Litte Log with the real error:
[log2.txt](https://drive.google.com/file/d/1Xj6CRJmGaGOYFaqCh2Ik3PkTKAdPx-OR/view?usp=sharing)

**Comment by AbdSattout** [comment]
Is TWRP crashing issue fixed?

**Reply by phhusson** [reply]
You're in the wrong place. This place isn't about TWRP, but Android ROMs

**Reply by AbdSattout** [reply]
Sorry, I mean after installing AOSP 12 gsi from TWRP... It crashes.
I remember that you said to someone in issues that the problem because of parsing storage.xml
_When booting to recovery and TWRP logo appears, the screen becomes black then the TWRP logo returns then becomes black... and still forever_
So is that issue fixed?

**Reply by phhusson** [reply]
That's a bug in TWRP.
But yes, as far as I know, TWRP fixed it back in december.

**Comment by tosuman** [comment]
Bluetooth headset is not working please fix the issue. Audio is not playing always stop you tube just paused and laggy also

**Comment by tosuman** [comment]
The Bluetooth issue is available in all GSI image Like Pixel Experience, AOSP also

**Reply by tosuman** [reply]
Sir , please resolve the bluetooth issue for the reason the hole experience got ruined

**Comment by ranmi91** [comment]
any chance they release a version for devices only a?
Apologies for my English, it's a Google translation

**Comment by mariasanchez84** [comment]
Amazing ! Works great thanks a lot.

**Comment by CessnaBroon** [comment]
@phhusson thanks for your hard work! Quick question regarding the OTA updater: what is it that quantifies a device as compatible? My Tab A (gta3xlwifi) messages

 "Error 
Your device does not support OTA updates!"

 when I attempt to enter the updater menu.
Why is this, and what can I do to make my device compatible?

File: system-squeak-arm32_binder64-ab-vndklite-vanilla-secure.img
I have opengapps 12L nano beta installed, along with Magisk 24.3.

Many thanks!

**Comment by wugo2021** [comment]
gsi12 on phones with android 8.0？

**Reply by Hausemaster8281** [reply]
I think we need a 4.19 kernel, I'm not sure but I think Android 8 didn't have 4.19 on any device, so, if I'm right these images can't work on it, but I'm not sure if devices with an ab partition with kernel below v4.xx can run these gsis at all...

**Reply by phhusson** [reply]
No you don't need 4.19 kernel.
This GSI boots just fine on Pixel 1 running Linux 3.18 kernel.

**Reply by Hausemaster8281** [reply]
Oh, okay, thank you very much for the info!

**Comment by antoniopedrocrs** [comment]
Hi sorry for my stupid question but  can i install on oneplus 6, enchilada? if yes, which file? thanks for the work. I currently have lineageos 18.1 installed.

**Reply by ghost** [reply]
Hello, if you phone is compatible with treble the only thing to do is check if if your device supports a/b scheme. If yes, go to wiki on this page, you'll see the differences between builds

**Comment by marth58** [comment]
every time google releases an update do you do it too or do you do it regardless of the errors it throws?

**Reply by Hausemaster8281** [reply]
I think so, as phh gives out a release with an SP and a lot of patches, and he releases more things if some major issues were being encountered...

**Comment by Theepaxio** [comment]
Hello, SMS are not working for me on this version ( Mi 11 Lite 5g ), they were working fine on previous versions.

**Comment by ghost** [comment]
Edit : it is not necessary to flash Gapps I guess! Anyway there is not enough space disk to do so.

@phhusson I have to thank you for your work, it is really appreciated !

**Reply by Hausemaster8281** [reply]
If you choose a variant with gapps then no

**Comment by SteQuOFFICIAL** [comment]
Anything known about the Google Pixel XL? Any changes that make a difference to previous builds?

**Comment by manosdoc** [comment]
Thanks for your hard and amazing work! This GSI boots and behaves excellent on the brand new Redmi Note 11 Global (spesn). ims works fine, wifi calling, volte. The only small bug is that tweaking the brightness slider at low totally blacks out the brightness.

**Comment by HansDus** [comment]
Just some Feedback for this Release on my Samsung Galaxy A505FN ...

It turned out much more stable then the last one (it did crash on me regulary) 
But it still crashes even if less now i hade to roll back to v313 witch works flawless

Saludos to @phhusson  from Hans D

**Comment by nation888** [comment]
Ref: Oukitel K9 Mediatek, TWRP 3.1.3 [system-squeak-arm64-ab-vndklite-gogapps-secure.img.xz]

I'm still on the V411 and still works the best so far. I tried installing V412 and couldn't get past the Google setup stuck on adding pin keeps repeating loop. I'm wondering if there is an vbmeta file needed or is there an encryption issue. Tried several times. Must be the security patch? Is there a zip file  that needs to be flashed along with the rom? Perhaps my old TWRP 3.1.3 needs to be kitchened to 3.6. (Tried  to build a newer 3.6 the file structures are somewhat different and when I merge the old stock rom missing files so haven't bothered flashing my frankensteined version yet.) If TWRP is the issue then I'll have to research down that path which is a rabbit hole I wanted to stay away from. 

1. Is it the Vbmeta?
2. Is it the TWRP version?
3. Is it the new Google encryption ?

**Comment by JonnyTech** [comment]
You do not need a custom recovery for flashing GSI's - fastboot is sufficient.

**Reply by Hausemaster8281** [reply]
You are right, but you sometimes need fastbootd on some devices though

**Comment by Theepaxio** [comment]
Kokote

Dne st 4. 5. 2022 15:36 uživatel Haus3master ***@***.***>
napsal:
